woocommerce/wc-admin
# Refactor and extract inbox panel components

Needed for https://github.com/Automattic/woocommerce-payments/issues/1662

The big changes in this PR:
- Refactored the inbox note and action button component to be more modular and less opinionated (I pushed several things up stream by the use of callbacks).
- Refactored the components to functional components and into TypeScript
- I moved most of the inbox panel components to the experimental package so it could be used for the issue linked above in WC Pay


### Detailed test instructions:

- Load this branch and go to **WooCommerce > Home**
- Finish the onboarding flow if on a fresh install
- Smoke test the inbox notes
     - Dismiss a single note
     - Dismiss all the notes
     - Check if action buttons work correctly
     - Check if the `wcadmin_inbox_note_view`, `wcadmin_inbox_action_dismiss` tracks are triggered.
